image/tools/entrypoint.sh

```diff
@@ -43,9 +43,14 @@ export HOME=$DEST
 component_dump_data $DEST
 echo '==> Component data dump completed'
 if [[ "$encryption_engine" ]]; then
-    encrypt_prepare $DEST
-    encrypted_files="$(encrypt_archive $ARCHIVES_DEST)"
-    echo '==> Data encryption completed'
+    check_encryption_enabled
+    if [[ $? -eq 0 ]]; then
+        encrypt_prepare ${DEST}
+        encrypted_files="$(encrypt_archive $ARCHIVES_DEST)"
+        echo '==> Data encryption completed'
+    else
+        echo "==> encryption secret not found. Skipping"
```

Comment:
@pb82 I might be missing something, but won't `encrypted_files` be undefined if we get to here. I think the S3 upload will fail?

Reply:
Good catch!